File: Array/Python/duplicate-element.py
# Approach 1: Using hashmap
# Time complexity is O(n)
class Solution:
    def containsDuplicate(self, nums) -> bool:
        hash_map = {}
        for i in nums:
            if i in hash_map:
                return True
            else:
                hash_map[i] = i
        return False

# Approach 2: Using Sets
# Time complexity is O(n)
class Solution:
    def containsDuplicate(self, nums) -> bool:
        if len(list(nums)) == len(set(nums)):
            return False
        else:
            return True

File: Array/Java/DuplicateElement.java
import java.util.HashSet;
import java.util.HashMap;

public class DuplicateElement {
    //using Hashset
    class SolutionSet {
        public boolean containsDuplicate(int[] nums) {
            HashSet<Integer> seen = new HashSet<>();
            for(int num:nums){
                if(seen.contains(num)){
                    return true;
                }
                seen.add(num);
    
            }
            return false;
        }
    }

    //Using Hashmap
    class SolutionMap {
        public boolean containsDuplicate(int[] nums) {
            HashMap<Integer, Integer> hashMap = new HashMap<>();
            for(int num:nums){
                if(!hashMap.containsKey(num)){
                    hashMap.put(num, 1);
                }
                else{
                    return true;
                }
            }
            return false;
        }
    }
}
